projects/extension/tests/test_litellm.py

```diff
@@ -0,0 +1,311 @@
+    if len(input_types) == 0:
+        pytest.skip("no input types for model")
+    for input_type in input_types:
+        cur.execute(
+            """
+            select vector_dims
+            (
+                ai.litellm_embed
+                ( %s
+                , 'hello world'
+                , api_key=>%s
+                , extra_options => jsonb_build_object('input_type', %s::text)
+                )
+            )
+            """,
+            (
+                name,
+                api_key_value,
+                input_type,
+            ),
+        )
+        actual = cur.fetchone()[0]
+        assert actual == dimensions
+
+
+@pytest.mark.parametrize(
+    "name,dimensions,api_key_name",
+    model_keys("name", "dimensions", "api_key_name"),
+    ids=ids,
+)
+def test_litellm_embed_with_multiple_inputs(
+    cur: psycopg.Cursor, name: str, dimensions: int, api_key_name: str
+):
+    api_key_value = os.getenv(api_key_name)
+    if api_key_value is None:
+        pytest.skip(f"environment variable '{api_key_name}' unset")
+    cur.execute(
+        """
+            select count(*) from ai.litellm_embed
+            ( %s
+            , ARRAY['hello world', 'hello bob']
+            , api_key=>%s
+            )
+        """,
+        (
+            name,
+            api_key_value,
+        ),
+    )
+    result = cur.fetchone()[0]
+    assert result == 2
+
+
+def load_secrets(*args):
+    secrets = {}
+    for name in args:
+        secrets[name] = os.getenv(name)
+        if secrets[name] is None:
+            pytest.skip(f"environment variable '{name}' unset")
+    return secrets
+
+
+def test_litellm_embed_azure_openai(cur: psycopg.Cursor):
+    secrets = load_secrets(
+        "AZURE_OPENAI_API_KEY", "AZURE_OPENAI_API_BASE", "AZURE_OPENAI_API_VERSION"
+    )
+    cur.execute(
+        """
+        select vector_dims
+        (
+            ai.litellm_embed
+            ( 'azure/text-embedding-3-small'
+            , 'hello world'
+            , api_key=> %s
+            , extra_options => jsonb_build_object(
+                'api_base', %s::text,
+                'api_version', %s::text
+              )
+            )
+        )
+        """,
+        (
+            secrets["AZURE_OPENAI_API_KEY"],
+            secrets["AZURE_OPENAI_API_BASE"],
+            secrets["AZURE_OPENAI_API_VERSION"],
+        ),
+    )
+    actual = cur.fetchone()[0]
+    assert actual == 1536
+
+
+def test_litellm_embed_bedrock(cur: psycopg.Cursor):
+    secrets = load_secrets(
+        "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_REGION_NAME"
+    )
+    cur.execute(
+        """
+        select vector_dims
+        (
+            ai.litellm_embed
+            ( 'amazon.titan-embed-text-v1'
+            , 'hello world'
+            , extra_options => jsonb_build_object(
+                'aws_access_key_id', %s::text,
+                'aws_secret_access_key', %s::text,
+                'aws_region_name', %s::text
```

Comment:
Is there any way to autoload these from env vars if they exist? same for the api_key